featuretools/entityset/entity.py

```diff
@@ -62,61 +62,39 @@ def __init__(self, id, df, entityset, variable_types=None,
                 instance across all child entities.
             encoding (str, optional)) : If None, will use 'ascii'. Another option is 'utf-8',
                 or any encoding supported by pandas.
-
+            make_index (bool, optional) : If True, assume index does not exist as a column in
+                dataframe, and create a new column of that name using integers the (0, len(dataframe)).
+                Otherwise, assume index exists in dataframe.
         """
         assert is_string(id), "Entity id must be a string"
         assert len(df.columns) == len(set(df.columns)), "Duplicate column names"
+        for c in df.columns:
+            if not is_string(c):
+                raise ValueError("All column names must be strings (Column {} "
+                                 "is not a string)".format(c))
+        if time_index is not None and time_index not in df.columns:
+            raise LookupError('Time index not found in dataframe')
+
+        created_index, index, df = create_index(index, make_index, df)
+        if index not in variable_types:
+            variable_types[index] = vtypes.Index
+
         self.data = {"df": df,
```

Comment:
can we avoid setting the data like this? perhaps by just making df an argument to some of the other functions? 

it feels wrong to set the data here and then call `self.update_data(...)` a second time at the end

Reply:
should we just call the relevant parts of `self.update_data` separately and skip calling `update_data`?.  It already expects the indexes to be set and it's a little strange to be "updating" the dataframe on initialization

Reply:
ya, i agree with that approach